api/views.py

```diff
@@ -1,3 +1,236 @@
+        user = request.user
+        try:
+            participant = models.Participant.objects.get(user=user)
+        except models.Participant.DoesNotExist:
+            return Response({'detail': 'Participant not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        if participant not in team.participants.all():
+            return Response({'detail': 'You are not in this team.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Ajouter le participant
+        team.participants.remove(participant)
+
+        # Vérifier si l'équipe est pleine
+        if team.participants.count() < team.hackaton.teamLimit:
+            team.full = False
+            team.save()
+
+        # Sauvegarder l'équipe après modification
+        team.save()
+
+        return Response(
+            {'message': f"Team {team.name} {team.code} updated successfully. Participant {user.first_name} left."},
+            status=status.HTTP_200_OK
+        )
+
+# class Hackaton
+class HackatonListView(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]  # Permet d'assurer que l'utilisateur est authentifié
+    queryset = models.Hackaton.objects.all()
+    serializer_class = serializers.HackatonSerializer
+
+class HackatonCreateView(generics.CreateAPIView):
+    permission_classes = [IsAuthenticated]  # Permet d'assurer que l'utilisateur est authentifié
+    queryset = models.Hackaton.objects.all()
+    serializer_class = serializers.HackatonSerializer
+
+    def create(self, request, *args, **kwargs):
+        # Sérialisation et création de l'objet
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            # Sauvegarder le hackathon
+            serializer.save()
+            return Response(
+                {'message': f'Hackathon {serializer.instance.name} created successfully.'},
+                status=status.HTTP_201_CREATED
+            )
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+# class Team
+class TeamCreateView(generics.CreateAPIView):
+    queryset = models.Team.objects.all()
+    serializer_class = serializers.TeamCreateSerializer
+
+    def create(self, request, *args, **kwargs):
+        # Sérialisation et création de l'objet
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            # Sauvegarder la team
+            serializer.save() 
+             # Retourner une réponse personnalisée
+            return Response(
+                {'message': f"Team {serializer.instance.name} {serializer.instance.code} created successfully."},
+                status=status.HTTP_201_CREATED
+            )
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class TeamDetailView(generics.RetrieveAPIView):
+    queryset = models.Team.objects.all()
+    serializer_class = serializers.TeamSerializer
+    lookup_field = 'id'
+
+class TeamHackatonListView(generics.RetrieveAPIView):
```

Comment:
you could've used a ModelViewSet, I believe it's less code to write

Reply:
didn't know about it